pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -2024,6 +2158,9 @@ class Build {
                         context.node(label) {
                             addNodeToBuildDescription()
                             // Cannot clean workspace from inside docker container
+                            if ( buildConfig.TARGET_OS == 'windows' && buildConfig.DOCKER_IMAGE ) {
+                                context.bat('rm -rf c:/workspace/openjdk-build/cyclonedx-lib c:/workspace/openjdk-build/security')
```

Comment:
this seems a little odd, why specific c:\workspace and why just cyclonedx-lib and security ?
presumably permissions? Why can't we just do    rm -rf ' + context.WORKSPACE + '/workspace

Reply:
This was the minimal that was required to clear up the workspace to allow the `git clean` operation to succeed without failures as I recall. I would definitely like to improve the way this is done, but those two directories specifically caused a problem (potentially because they are written to during the build, and that happens inside the docker container under different permissions?) I would strongly support a separate issue being raised to look at whether we can avoid writing into that directory and/or move/change the scope of this particular `rm` operation. We do have various worksapce clean-up options on the build to clear things up, but unless that is done at the correct time (as per the comment above the bit you've highlighted) it doesn't work in the container case.

That was a longer answer that I planned, but let me know if you're ok with deferring that to an action after this PR, or if you want it addressed before you feel able to approve it. I don't like it, but I'm also conscious of the fact that the PR as-is has undergone a good amount of testing for now and there are multiple cases that would have to be tested to fully test additional changes here.

Reply:
I have done some tests without this rm and so far not reproduced the failure but I suspect it's an edge case on certain failures which triggered the value to reset the workspace so I'm still more comfortable leaving this in if possible. But definitely worth keeping an eye on so I could add some debug in to keep an eye on it and see later on if it might have failed without it (I'm running with that in another branch for the testing)

Reply:
ok makes sense as an edge case, but can we use:
```
'rm -rf ' + context.WORKSPACE + '/cyclonedx-lib'
'rm -rf ' + context.WORKSPACE + '/security'
```

Reply:
Happy to change it, but any particular reason? Just feels more efficient to spawn one shell and rm process instead of two

Reply:
At that point WORKSPACE is resolving to `c:\workspace\workspace\build-scripts\jobs\jdk21u\...` instead of the "shortname" so it would clean up the wrong directory (Also it's using `\` instead of `/` which is likely to cause problems)

Reply:
OK I can get it to use that variable if I move the `workspace` definition earlier in the `build()` function and remove the multiple definitions, then wrapping this part in `context.ws(workspace)`

Reply:
Pushed the change and seems to work in the docker environment on initial tests - LMK if moving the scope of the variable is likely to cause any side-effects e.g. on other platforms..

Reply:
let me check, didn't realize this was out of scope of the context.ws()

Reply:
yeah looks good now, buildScripts() is run within the context.ws()...